lib/smartcar.rb

```diff
@@ -39,45 +25,154 @@ class AuthenticationError < ExternalServiceError; end
   class BadRequestError < ExternalServiceError; end
 
   # Host to connect to smartcar
-  SITE = 'https://api.smartcar.com/'
+  API_ORIGIN = 'https://api.smartcar.com/'
+  PATHS = {
+    compatibility: '/compatibility',
+    user: '/user',
+    vehicles: '/vehicles'
+  }.freeze
 
   # Path for smartcar oauth
-  OAUTH_PATH = 'https://connect.smartcar.com/oauth/authorize'
+  AUTH_ORIGIN = 'https://connect.smartcar.com'
   %w[success code test live force auto metric imperial].each do |constant|
     # Constant to represent the value
     const_set(constant.upcase, constant.freeze)
   end
 
-  # Lock value sent in request body
-  LOCK = 'LOCK'
-  # Unlock value sent in request body
-  UNLOCK = 'UNLOCK'
-  # Start charge value sent in request body
-  START_CHARGE = 'START'
-  # Stop charge value sent in request body
-  STOP_CHARGE = 'STOP'
   # Constant for units
   UNITS = [IMPERIAL, METRIC].freeze
 
-  # Smartcar API version variable - defaulted to 1.0
-  @api_version = '1.0'
-
-  # rubocop:disable Naming/AccessorMethodName
-  # Module method Used to set api version to be used.
-  # This method can be used at the top to set the version and any
-  # following request will use the version set here unless overridden
-  # separately.
-  # @param version [String] version to be set without 'v' prefix.
-  def self.set_api_version(version)
-    instance_variable_set('@api_version', version)
-  end
+  # Smartcar API version variable - defaulted to 2.0
+  @api_version = '2.0'
+
+  class << self
+    # Module method Used to set api version to be used.
+    # This method can be used at the top to set the version and any
+    # following request will use the version set here unless overridden
+    # separately.
+    # @param version [String] version to be set without 'v' prefix.
+    def set_api_version(version)
+      instance_variable_set('@api_version', version)
+    end
+
+    # Module method Used to get api version to be used.
+    # This is the getter for the class instance variable @api_version
+    #
+    # @return [String] api version number without 'v' prefix
+    def get_api_version
+      instance_variable_get('@api_version')
+    end
+
+    # Module method Used to check compatiblity for VIN and scope
+    #
+    # API Documentation - https://smartcar.com/docs/api#connect-compatibility
+    # @param vin [String] VIN of the vehicle to be checked
+    # @param scope [Array of Strings] - array of scopes
+    # @param country [String] An optional country code according to
+    # [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2).
+    # Defaults to US.
+    # @param options [Hash] Other optional parameters including overrides
+    # @option params [String] :client_id Client ID that overrides ENV
+    # @option params [String] :client_secret Client Secret that overrides ENV
+    # @option params [String] :version API version to use, defaults to what is globally set
```

Comment:
should `flags` option be documented here too?

Reply:
and used....i dont think i am using it right now.